lib/src/model/analysis/analysis_controller.dart

```diff
@@ -46,6 +49,7 @@ class AnalysisOptions with _$AnalysisOptions {
     int? initialMoveCursor,
     LightOpening? opening,
     Division? division,
+    @Default(false) bool isBroadcast,
```

Comment:
I think we should not re-use the analysis controller for broadcast games.

I know there is a lot of logic in common, but still, they have complete different purposes and enough differences so that we should split them.

For now I'd say let's copy/paste all the logic we need from analysis controller into `BroadcastGameController`. We'll figure out later how to factor out common logic properly. 

I said the same thing to @tom-anders about study controller. I guess both will share a good amount of logic too.

(One thing we can extract now I think, is the engine preferences. We should separate what is purely engine settings (nb of threads, memory, etc) from the analysis settings.)

Reply:
I believe I need all the logic of the analysis controller except maybe the method to request a server analysis (I don't know if there are people who can ask a server analysis for a broadcast game or if it is automatic). Am I missing some other parts that I do not need for the broadcast analysis controller ?

Reply:
Right, I think you need almost everything. I just don't want to add methods and logic belonging to broadcast (or any other feature) to the analysis controller. If we continue to do that for all features (study, explorer, broadcast, etc.), it can become a mess I fear.
Also we shouldn't use `standaloneBroadcastId`. A broadcast is never standalone, but I understand why you chose this approach, it was the simplest way to integrate with the analysis controller.

So you can leave it like that for now, let's focus on getting the broadcast feature out. I'll figure later how to refactor this.